### PR TITLE
Remove cold methods from the CPU cache

### PR DESCRIPTION
**Pull request by gdkchan**
Currently translated methods are stored on the cache and never removed, which creates an unbounded memory growth. This attempts to improve that by removing old methods similar to what the OGLCachedResource does. Since the TryGetSubroutine method is hot I tried to avoid doing too much work on it, it only update the timestamp and the linked list when the call count exceeds a given threshold. It was not throroughly tested and may need tweaking.

### CODE REVIEW

**Comment by ReaperOfSouls1909** [comment]
capture
stardew valley no longer shows anything on boot 

its the only effect game out of all i have

**Comment by gdkchan** [comment]
Works fine on my end.

File: ChocolArm64/ATranslatorCache.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool TryGetSubroutine(long Position, out ATranslatedSub Subroutine)
+        {
+            if (Cache.TryGetValue(Position, out CacheBucket Bucket))
+            {
+                if (Bucket.CallCount++ > MinCallCountForClear)
+                {
+                    Bucket.CallCount = 0;
```
[the remark this answers is not in the document]
Comment (gdkchan) [reply ChocolArm64/ATranslatorCache.cs]: Thanks, this should be indeed more efficient, I updated it.

**Comment by gdkchan** [comment]
Rebased.

**Pull request merged**